ng2-components/ng2-alfresco-core/src/services/upload.service.ts

```diff
@@ -183,7 +184,7 @@ export class UploadService {
             emitter.emit({ value: data });
         })
         .catch(err => {
-            this.onUploadError(file, err);
+            return err;
```

Comment:
`.catch()` returns a promise. The only problem I see with this statement is that a later `.then()` will be trigger and not a `.catch()`. We should `throw err`.